mobile/library/common/engine_types.h

```diff
@@ -32,6 +36,40 @@ struct EnvoyEventTracker {
   absl::AnyInvocable<void()> on_exit_ = [] {};
 };
 
+/** The Envoy error passed into `EnvoyStreamCallbacks::on_error_` callback. */
+struct EnvoyError {
+  envoy_error_code_t error_code;
+  std::string message;
+  absl::optional<int> attempt_count = absl::nullopt;
+  absl::optional<std::exception> cause = absl::nullopt;
+};
+
+/** The callbacks for the stream. */
+struct EnvoyStreamCallbacks {
+  absl::AnyInvocable<void(const Http::ResponseHeaderMap&, bool /* end_stream */,
```

Comment:
should we preserve the prior functions around what each of these callbacks entail?

Reply:
Comments added.